plugin/pkg/proxy/connect.go

```diff
@@ -117,6 +119,31 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 	for {
 		ret, err = pc.c.ReadMsg()
 		if err != nil {
+			if proto == "udp" {
```

Comment:
`proto` isn't necessarily reliable to determine the actual protocol used with the upstream host.  For example, when forwarding UDP requests to upstream over DoT (dns over tls) connections, the value of proto will be "udp" here, while the actual protocol is over tcp.

Reply:
oh ok understood. Changed the condition here - if p.transport.transportTypeFromConn(pc) == typeUDP